post.md

```diff
@@ -0,0 +1,269 @@
+---
+post_title: ssh reverse tunnel for PXE, NFS and DHCP setup on QubesOS
+author: Piotr Król
+post_excerpt: ""
+layout: post
+published: true
+tags:
+  - QubesOS
+  - networking
+  - pxe
+  - nfs
+  - iptables
+  - ssh
+categories:
+  - OS Dev
+---
+
+At some point I stuck in the forest with WiFi connection and no physical access
+to router to create nice networking for my coreboot development needs. Recently
+I switched my laptop to QubesOS what give interesting flexibility, but also
+additional problems.
+
+My key requirement is to boot system over PXE, so I can easily do kernel
+development and play with Xen. Because only available connection for my apu2
+platform was directly to my laptop I had to provide configured DHCP server and
+PXE server on it. QubesOS networking is quite complex and to get to VM you have
+to pass-through at least `sys-net` VMs. Those VMs requires `iptables`
+configuration to correctly pass traffic or some tricks as I presented below.
+
+I don't think much people will face so weird configuration, but I need below
+notes for myself and there is some chance that someone will face similar
+issues.
+
+To summarize my target configuration was like that:
+
+![qubes-apu2-setup](https://3mdeb.com/wp-content/uploads/2017/07/qubes-apu2-setup.png)
+
+My initial idea was to have servers on AppVMs, but I didn't have enough time to
+get through QubesOS `iptables` rules. What lead to discover interesting
+alternative with `proxychains`.
```

Comment:
The thread is cut all of a sudden here. Maybe consider adding something like: "... proxychains, which I will describe later in this article".

Reply:
ok